codechecker_common/plist_parser.py

```diff
@@ -243,45 +243,53 @@ def fids_in_edge(edges):
     return fids
 
 
-def fids_in_path(report_data, file_ids_to_remove):
+def transform_diag_element(element, file_ids_to_remove, new_file_ids):
     """
-    Skip diagnostic sections and collect file ids in
-    report paths for the remaining diagnostic sections.
+    This function will update every file attribute of the given element by
```

Comment:
Please add some more documentation about what an element can be. It should be a diagnostic section right?
Shouldn't that always be a dict? When will it be a list or tuple?

Reply:
I updated the documentation.

On the first call it should be a diagnostic section. It will recursively traverse all children of it.
It can be a list for example when parsing notes:
```xml
<key>notes</key>
<array>
</array>
```
or edges:
```xml
<key>edges</key>
<array>
</array>
```